blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
After thinking about it a bit more I'm starting to think the same. We need to remember to put a warning about this somewhere in the tutorials though, because it can be very counter-intuitive if you don't fully realize this limitation.